sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs

```diff
@@ -78,7 +80,17 @@ private string GetSessionFilePath()
             string className = testAdapter.ClassName.Substring(testAdapter.ClassName.LastIndexOf('.') + 1);
 
             string fileName = name + (IsAsync ? "Async" : string.Empty) + ".json";
-            return Path.Combine(TestContext.CurrentContext.TestDirectory,
+
+            string path = ((AssemblyMetadataAttribute) GetType().Assembly.GetCustomAttribute(typeof(AssemblyMetadataAttribute))).Value;
+
+            if (path == null) // send to test directory
```

Comment:
Why do we need this fallback? The attribute will be in every project. If not populated, should we just throw?